### PR TITLE
Pract.05: Add 01-08 | Mihail Dimitrov

### PR DESCRIPTION
**Pull request by MihailDimitrov1235**
Mihail Dimitrov
3 3MI0600472

### CODE REVIEW

File: Sem.05/Pract.05/MihailDimitrov/Task01.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+using namespace std;
+
+void average(int arr[], int n) {
```
Comment (SirLex) [inline Sem.05/Pract.05/MihailDimitrov/Task01.cpp:4]: We accept arr as a logical name for a variable, I can not say the same thing for n -> try to use meaningful namings for your variables, sizeOfArray? countOfElements?

File: Sem.05/Pract.05/MihailDimitrov/Task01.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	int closest = arr[0];
+
+	for (int i = 1; i < n; i++)
+	{
+		if (abs(average - arr[i]) < abs(average - closest)) {
+			closest = arr[i];
+		}
+	}
+
+	cout << closest;
```
Comment (SirLex) [inline Sem.05/Pract.05/MihailDimitrov/Task01.cpp:21]: We should strive to have a single responsibility functions -> have a separate function for finding closest, and a separate for average. One function should have only 1 responsibility.

File: Sem.05/Pract.05/MihailDimitrov/Task01.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+}
+
+int main()
+{
+	int n;
+	cin >> n;
+
+	int arr[1000];
```
Comment (SirLex) [inline Sem.05/Pract.05/MihailDimitrov/Task01.cpp:30]: We should use constants and not magical numbers such as 1000.

File: Sem.05/Pract.05/MihailDimitrov/Task01.cpp (hunk 1 above)
Comment (SirLex) [inline Sem.05/Pract.05/MihailDimitrov/Task01.cpp:4]: Also, why not return the average? We should strive further from void functions. When we have some return type we know what to expect from a function plus it is easier to test as you will see in the following years

File: Sem.05/Pract.05/MihailDimitrov/Task02.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+using namespace std;
+
+void minAndMax(int arr[], int n) {
+
+	if (arr == NULL)
+	{
+		return;
+	}
```
Comment (SirLex) [inline Sem.05/Pract.05/MihailDimitrov/Task02.cpp:9]: good catch! I like it

File: Sem.05/Pract.05/MihailDimitrov/Task05.cpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
+#include <iostream>
+using namespace std;
+
+bool isSymetric(int arr[], int n) {
+	if (arr == NULL)
+	{
+		return false;
+	}
+
+	int j = 0;
```
Comment (SirLex) [inline Sem.05/Pract.05/MihailDimitrov/Task05.cpp:10]: not used variable.

File: Sem.05/Pract.05/MihailDimitrov/Task08.cpp [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+{
+    if (arr == NULL)
+    {
+        return;
+    }
+
+    for (int i = 0; i < n; i++)
+    {
+        bool containsIndex = false;
+        int el = arr[i];
```
Comment (SirLex) [inline Sem.05/Pract.05/MihailDimitrov/Task08.cpp:14]: el ? as element? write it out it is easier to read and understand the code that way.

**Review by SirLex — COMMENTED** [review]
Overall I am very happy with the tasks! Good job.
Few notes will be to aim to write code such that others can understand easily!
Minor comments, but still great!

**Review by AtanasHrist0v — CHANGES_REQUESTED** [review]
Please review the instructions [here](https://github.com/SirLex/UP/blob/main/Git%20&%20GitHub%20Tutorials/README.md), point 5, last bullet point.

**Pull request merged**